### ISSUE TITLE
fresh() doesn't call any lifecycle Hook

### ISSUE THREAD

**Issue by abd777** [CodeDredd/pinia-orm#1722]
### Reproduction

[Playground](https://pinia-orm-play.codedredd.de/#N4IgDghgxg1hDmBTAziAXAbVAOwgW0XRADcBXRAWgBNE8BLEAGhGQHtSAnKQtEU7MDHhMQNZFA50wAFzqtsRAGrkABABFadFQAoAFhAA2AMxV1sKgGIdE2KLoCUp5CogqaxRAdZgz8FaVkDOmkATxcDLwB3ZxD2FWlWFUjJaUQVAFVsOg8OZEMVZTSAGToAIw4ISRQVI1YOAtUAJhUAMhUAZhFOAyJdaWkwZDQAemH+QXgAOihWPGGyShp6YaDS4bMaAA9JvAArVABfRhx8HhAAAQXSZERh5H1rKhE2Tm4icaERMQkpWXkiADKD0QVAaiHSN38gWCdBQky6HB6vD6AyGow+Uxmc0u5Gut3ulRB5wArJMAAyTACM62wW0mKDwk32ICOJwIRBxiDxwxm1me7C4ZwxXxQPxkcgUvAAwqwIogoH9zKwTCgbthZPlCioZXgwKxkDD5BlobIUAikSAUYMRmMBEJprNhpzubzECTyVSaXSGUzDscQLh2bxOcN3AlZcgKBAfPzXkK7cJmN9JOL-rwAJLqxAVBVJYK6eK6NJasOsCM1DizQtpSBIc29frW9EJh3YhahxDEcMGSPRujnABsk0antWwwZXsQ22ZrIDpyIPiyEFjgveCZF4hTiqIAAUzHRl8xug3UTaMa3houD+cR5TJgPQ3RkNJL-uIPTkHgKOVWNFs76WX9QMznOK8IAoOo5mwSDDDoAAvDgVzeXhhSTUUtwlXc3xUAB5AAlABZFRoI4PBYIQ+tkUbNFbQmC9QLfCDSOGEiyKCBDzjvAB2clH2fSdtj2P02TOXBoLoKgWPkCgbigThCGYF5VxQ9c0M3X5MN4AA5CBxKeI9ERPJtaPtLEWN01gJPOAAWD0ySk7AZPleSBIA2dgIXRjIKQ+MJg3MVt14Pcl1wwjKMtaizxbMywKYvBOMmHj7KoJ8Xw2KcdhnABdZgjDoAwzTQUAAEEwDASYFnQUBUl1AwIFSIgbjSGZsDYArngCog6F1OppBUYAVGsIxGBUeQCPYLNQQOCsqwAcgWWaAB0sh6jg+oGvE8MQPUVGmoxKzwFRZtiyClpWvU1oyG56n2ubJmGCF-32M6VGW7qLr6qh6tcW7Dtm+6vukd9nuW17bHkZ9-GuradoAXihxAYdYbRHo4exQeWlrIbxDgAU8eU+vhobtGwUgInR7BluWsaJtSKhtG0ZbHFhgA+frQZULG+px5x4ZxpHJkifNtFmhIqH1Wb7EmJBpG0CmVE5iHZUQSYvHgEWAGIte1nWtYVlQAWKxR0y0gBxDIAQAUTwhXdbt7XJY5-nttYSY8g8bRAffHmKY5rHldV1h1dmu2DaNy21BUO3HfMRXWoDtWRYAPxT1O09TlR06ztOY4V-2CsD4P7b1vDLZ3IpiqlE3zfSK2beLrXc4RjgBf2lBdE977Jh9v2lYLxOQ+LlRS-LyuI6jhuY4OX2FCOPhrqlSoqAq8gqpAGqwDqhreDx5q+4UlhOt4FRANELvmSK9faE3+qzg6rciCnD63EQIwIDJ9aneuoYVAwDmFeAP-fWEk0AqEpIwIBCtgKgNmgAKVYLobAABpWaEDY76xcMQb6HB0x4HgOkREMCrQ0ToJMMAFQsFAw4NMKAwx2hkjJAAfm6vAWGxIuKoMgTUOguRpA6QIDA+BiDOHoP1nVZ8-DEAwJQWgjBCsxb6lAX-URGDAEqLkbIaQBUYEAlIGAbMIi5H6yOFwgBpj9aaO0UdGUspDFGN2rIuROUgEmNEWooxICVCNEcRg6BR1YG6UQGoOxGCICUMqLg-BhCjrEJtKQ8hYTsE0LoQw5heDYbWWJCE-WeVeGSMEYE7JCtxF8NODAtQrBEBFPiKwcWP9lH2PZuojBlipFHS1O0Jwbh5BVJ8XI1xjT3GNPiMEKxs0toFQgJCLSpBNh9XaNUhWAz7FDMaa0mBxUqCgmfKQIwRhFkOK4c40RBwOZZWWgcU+qMAKXw3lvO+ikj4gHer1fqKhxo0AMLtGaf0TqkTOi8y6AAVWpiRfpHXuiC8WvozrLSfq8mgb8P6c3Ec4VGKgpypFpM4D5ngmkK2fPVOgUAMXqmCGEeGs0eawtjoS2QJK8qeCoMgOW+L9bWGkJwcwqzgFUFAdIXQT5u4STln0qBZTCxCufJIbAwdJZiswdgyJBCDD8sFcgN20gZVyvsAq3JEiJUCqlVq3wIt5VcJKfkyVGrpWmtmua9RCif5Go1foZABFdIhG0FC1gI0qXXXTFQB1GDTmx1DRzOlxLObWCJbK7QeBameEcGooB+cVYD0WiACQiBY3wEzSoNALM2ZN1DQSoG9Lo05rpvGxNBhk1U1EWmwuItM3ZtvlQfNhbi3y12g2stRKSV6MBqahNnz62U0bfvZtS0+BgGHbKztRajo9tLSoSNg653tprWOppqap0ZtnYDEEi7u0c3DbS8tUbPmIFkHG0dSbd2Tvjv3IOLbRD41vXmkABal0lojZekl17q33rrY+jBTaD1AePd+rty6z19rXQBtdYSR21vHXu596bX0zoBChhdMHf0rv-QO5DHh6YgZGtYXkVBx3gf3dhzN7toMKy7THeDE6Lmnx9Tc6q197lEHvr8Lqq11rvNrd88Fx0vL-IbfCy6iL34GD6lAVFKgfUYrmTYZlYnPlsvXaSzRFKjpOppf2itjKDDMtZTyjlXK2UYM8S64V9NdVcPWdazV2qzWufUTjQNaqhX1S1STMmda+mrtDZcg40WgA)

### Describe the bug

I need to listen for the lifecycle hook inside the `Model` when the `fresh()` is called , but no any lifecycle hook is listening to `fresh()` while if i use `save()` it works

For now i'm using this as a workaround 
```
const repo = useRepo(REPO_NAME)
// clear everything
repo.flush()
// then save , i.e to trigger the lifecycle
repo.save(NEW_DATA)
```
ideally `fresh()` should also be a lifecycle hook trigger

**Comment by harmyderoman** [comment]
Actually, it can be useful, if you have one mutator that doesn't trigger the hooks. For example, if you need to save raw data from a backend.

**Comment by abd777** [comment]
@harmyderoman  I agree with you , But the issue is that in most cases we need to check the hooks ( as we're using `saving`  
 and other hooks for other functions )  
 
 May be we can add an optional `raw` parameter like  `{raw:true}` as the last config object and if `raw` is true we can keep it as it is , otherwise we can listen for the hooks

**Comment by CodeDredd** [comment]
Hmmm interesting. I try to get my thoughts around that. You both have valid points.
I think flush is getting hooks and i also add the option to avoid hooks.

**Comment by groenroos** [comment]
Ran into this today - I wanted to use lifecycle hooks to normalise & process raw data from the backend into a predictable and useful format, so it was unexpected that certain insertion methods simply ignore all hooks. From this perspective, IMHO this behaviour feels like a gotcha, and almost even defeats the very purpose of hooks.

I can definitely understand a use case for wanting to initiate the store with data without firing hooks - but this feels like something you should explicitly express during the action (such as the suggested `{raw: true}` or similar), and not inherent behaviour you need to know to override or avoid.

In the interim, this note should be added to the documentation, because neither the `Inserting Data` nor the `Lifecycle Hooks` pages describe this behaviour, and I only learned about it from this issue after spending slightly too long debugging 😅

**Comment by CodeDredd** [comment]
Implemented in #2036 (v2.0.0, pending review): `fresh()` now fires `saving`/`creating` before and `saved`/`created` after persisting, per record. A before hook returning `false` excludes that record. If you want the old raw behaviour (e.g. seeding the store without hooks), pass `fresh(data, { raw: true })` — the opt-out flag suggested here. The docs now also state explicitly which methods fire which hooks.
